File: 797-rabbits-in-forest/rabbits-in-forest.js
var numRabbits = function(answers) {
    const frq = {};
    for (const a of answers) 
        frq[a] = (frq[a] || 0) + 1;
    
    let ans = 0;
    for (const k in frq) 
        ans += Math.ceil(frq[k]/ (parseInt(k) + 1)) * (parseInt(k) + 1);

    return ans;
};

File: 797-rabbits-in-forest/rabbits-in-forest.java
class Solution {
    public int numRabbits(int[] answers) {
        Map<Integer, Integer> mp = new HashMap<>();
        for (int i : answers) mp.put(i, mp.getOrDefault(i, 0) + 1);
        int ans = 0;
        for (Map.Entry<Integer, Integer> p : mp.entrySet())
            ans += Math.ceil((double)p.getValue() / (p.getKey() + 1)) * (p.getKey() + 1);

        return ans;
    }
}